lib/active_merchant/billing/gateways/securion_pay.rb

```diff
@@ -79,6 +79,29 @@ def verify(credit_card, options={})
         end
       end
 
+      def store(credit_card, options = {})
+        if options[:customer_id].blank?
+          MultiResponse.run() do |r|
+            #create charge object
+            r.process { authorize(100, credit_card, options) }
+            #create customer and save card
+            r.process { create_customer_add_card(r.authorization, options) }
+            #void the charge
+            r.process(:ignore_result) { void(r.params["metadata"]["chargeId"], options) }
+          end
+        else
+          verify(credit_card, options)
```

Comment:
When provided a valid customerId, the act of creating a new charge automatically adds the card to the customer (relevant paragraph from api doc pasted below). The authorize stage of verify does that job. Although subsequent void refunds the charge, the card stays added to the customer since the authorize step was successful. The 3rd test-case within 'test_successful_store' verifies this. Verify response is pasted further down below. The newly added card's id is available within the response params.

> customerId
> string, optional (either customerId or card is required)
> Identifier of customer that will be associated with this charge.
> This field is required if charge is being created with customer's existing card.
> If specified then successful charge created with new card (either by token or by card details) will add that card to customer's card and is will be set as customer's default card.

`{"id":"char_fIZbI9jfq4t6TeE1n0o0fe6X","created":1458839587,"objectType":"charge","amount":100,"currency":"USD","description":"ActiveMerchant test charge","card":{"id":"card_JmBx1cu2yeFxdk6YIWMrUURd","created":1458839587,"objectType":"card","first6":"424242","last4":"4242","fingerprint":"ecAKhFD1dmDAMKD9","expMonth":"9","expYear":"2017","cardholderName":"Longbob Longsen","brand":"Visa","type":"Credit Card"},"captured":false,"refunded":false,"disputed":false,"metadata":{"email":"foo@example.com"}}`

`   @primary_response=
    #<ActiveMerchant::Billing::Response:0x007ffce4123b38
     @authorization="char_fIZbI9jfq4t6TeE1n0o0fe6X",
     @avs_result=
      {"code"=>nil, "message"=>nil, "street_match"=>nil, "postal_match"=>nil},
     @cvv_result={"code"=>nil, "message"=>nil},
     @emv_authorization=nil,
     @error_code=nil,
     @fraud_review=nil,
     @message="Transaction approved",
     @params=
      {"id"=>"char_fIZbI9jfq4t6TeE1n0o0fe6X",
       "created"=>1458839587,
       "objectType"=>"charge",
       "amount"=>100,
       "currency"=>"USD",
       "description"=>"ActiveMerchant test charge",
       "card"=>
        {"id"=>"card_JmBx1cu2yeFxdk6YIWMrUURd",
         "created"=>1458839587,
         "objectType"=>"card",
         "first6"=>"424242",
         "last4"=>"4242",
         "fingerprint"=>"ecAKhFD1dmDAMKD9",
         "expMonth"=>"9",
         "expYear"=>"2017",
         "cardholderName"=>"Longbob Longsen",
         "brand"=>"Visa",
         "type"=>"Credit Card"},
       "captured"=>false,
       "refunded"=>false,
       "disputed"=>false,
       "metadata"=>{"email"=>"foo@example.com"}},
     @success=true,
     @test=true
`